@yxhuvud this is an example of a compilation buffer's contents in a project I'm working on. Note that the first error message starts with "Error in". You can paste this text in a buffer and activate `compilation-mode`. (The default-directory won't make sense on your platform, nor will the file names of course.)

```
-*- mode: compilation; default-directory: "~/src/github/cryo_master/src/cryo_master/" -*-
Compilation started at Wed May 15 11:45:29

makeup build
shards build
Dependencies are satisfied
Building: cryo_master
Error target cryo_master failed to compile:
Error in src/cryo_master.cr:94: instantiating 'CryoMaster:Module#run()'

CryoMaster.run
           ^~~

in src/cryo_master.cr:61: instantiating 'Main#run()'

      Main.new(cm).run
                   ^~~

in src/cryo_master/curses/main.cr:37: instantiating 'loop()'

      loop do
      ^~~~

in src/cryo_master/curses/main.cr:37: instantiating 'loop()'

      loop do
      ^~~~

in src/cryo_master/curses/main.cr:45: instantiating 'CM#prev_patch()'

            @cm.prev_patch
                ^~~~~~~~~~

in src/cryo_master/cm.cr:154: instantiating 'prev_patch((Song | Nil))'

    prev_patch(@song)
    ^~~~~~~~~~

in src/cryo_master/cm.cr:165: undefined method 'patches' for Nil (compile-time type is (Song | Nil))

      @patch = @song.patches[(@song.patches.index(@patch) || -1) - 1]
                     ^~~~~~~

Rerun with --error-trace to show a complete error trace.

make: *** [build] Error 1

Compilation exited abnormally with code 2 at Wed May 15 11:45:30
```

--- a/crystal-mode.el
+++ b/crystal-mode.el
@@ -2762,7 +2762,7 @@ directory of the current file."
            (boundp 'compilation-error-regexp-alist-alist))
   (add-to-list 'compilation-error-regexp-alist 'crystal-spec)
   (add-to-list 'compilation-error-regexp-alist-alist
-               '(crystal-spec . ("^(Error )?in \\([^()\t\n]+\\):\\([0-9]+\\):? .*$" 2 3)) t)))
+               '(crystal-spec . ("^\\(Error \\)?in \\([^()\t\n]+\\):\\([0-9]+\\):? .*$" 2 3)) t)))
 
 ;;; Invoke crystal-mode when appropriate
 

